astropy/io/fits/header.py

```diff
@@ -2285,3 +2317,18 @@ def _check_padding(header_str, block_size, is_eof, check_block_size=True):
         actual_len = len(header_str) - block_size + BLOCK_SIZE
         # TODO: Pass this error to validation framework
         raise ValueError(f'Header size is not multiple of {BLOCK_SIZE}: {actual_len}')
+
+
+def _hdr_data_size(header):
```

Comment:
I didn't name it _data_size since that is an existing attribute of HDU. However, it won't do harm having the same name. Let me know if I should change it.